src/c4/runtime_check.cc

```diff
@@ -0,0 +1,36 @@
+//----------------------------------*-C++-*----------------------------------//
+/*!
+ * \file   ds++/runtime_check.cc
+ * \author Kent Grimmett Budge
+ * \date   Wed Mar 28 07:58:48 2018
+ * \brief  Member definitions of class runtime_check
+ * \note   Copyright (C) 2018 Los Alamos National Security, LLC.
+ *         All rights reserved. */
+//---------------------------------------------------------------------------//
+
+#include <exception>
+
+#include "C4_Functions.hh"
+#include "ds++/DracoStrings.hh"
+
+namespace rtt_dsxx {
+
+//-------------------------------------------------------------------------//
+void runtime_check(bool condition, char const *message) noexcept(false) {
+  Require(message != nullptr);
```

Comment:
Does it make sense to make this function header-only and inline?

Reply:
There's enough going on that I hesitate. Since this function is not suitable for calling within loops or other compute-intensive code (because it must be synchronous) it's not compute critical. The construction of the throw message takes some code.

Reply:
I'll leave that decision up to you.  Anyway, it is doubtful that the function would ever be inlined because of the embedded `throw`.